Cookie Store: Add testing for checks on path and domain length when setting cookies

RupinMittal opened the pull request:
The spec (https://wicg.github.io/cookie-store/#set-cookie-algorithm) for the Cookie-Store API dictates that in the set function, if the byte sequence length of the path (in UTF8 format) is greater than the maximum attribute value size (currently 1024 bytes according to https://wicg.github.io/cookie-store/#cookie-maximum-attribute-value-size), then the promise should be rejected with a TypeError. The same is true for the domain. This patch adds testing for this check in the set function in the same file that tests other argument-related checks that the set function should be doing.

pwnall commented on cookie-store/cookieStore_set_arguments.https.any.js:
```diff
@@ -285,3 +285,29 @@ promise_test(async testCase => {
+  await promise_rejects_js(testCase, TypeError, cookieStore.set(
+      { name: 'cookie-name',
+        value: 'cookie-value',
+        domain: '/this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-t ...
```
Make this a valid domain, so the only error is the length limit.

RupinMittal replied:
Good point. Fixed!

pwnall commented on cookie-store/cookieStore_set_arguments.https.any.js:
```diff
@@ -285,3 +285,29 @@ promise_test(async testCase => {
+  await promise_rejects_js(testCase, TypeError, cookieStore.set(
+      { name: 'cookie-name',
+        value: 'cookie-value',
+        path: '/this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-thi ...
```
Checking the length assumption here is a bit inconvenient.

Consider a more obvious formulation, such as `'/' + 'a'.repeat(1024)`

RupinMittal replied:
Good point. Fixed!

inexorabletash commented on cookie-store/cookieStore_set_arguments.https.any.js:
```diff
@@ -285,3 +285,29 @@ promise_test(async testCase => {
+
+promise_test(async testCase => {
+  // The maximum attribute value size is specified at https://wicg.github.io/cookie-store/#cookie-maximum-attribute-value-size.
+  const maximumAttributeValueSize = 1024
```
This isn't used. Remove it?

RupinMittal replied:
Good point. Fixed!

inexorabletash commented on cookie-store/cookieStore_set_arguments.https.any.js:
```diff
@@ -285,3 +285,29 @@ promise_test(async testCase => {
+
+promise_test(async testCase => {
+  // The maximum attribute value size is specified at https://wicg.github.io/cookie-store/#cookie-maximum-attribute-value-size.
+  const maximumAttributeValueSize = 1024
```
This isn't used. Remove it?

RupinMittal replied:
Good point. Fixed!

pwnall reviewed (changes requested):
Please ask @ayuishii for a general review here. If I'm still in some OWNERS file, please replace me with @ayuishii :)

I hope these comments help!

inexorabletash commented on cookie-store/cookieStore_set_arguments.https.any.js:
```diff
@@ -285,3 +285,29 @@ promise_test(async testCase => {
+  await promise_rejects_js(testCase, TypeError, cookieStore.set(
+      { name: 'cookie-name',
+        value: 'cookie-value',
+        path: '/this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-this-string-has-1025-characters-in-it-thi ...
```
Rather than hardcoding, might be better to construct this e.g. `'/' + 'x'.repeat(1023) + '/'`

(And then you could use the otherwise unused constant.)

RupinMittal replied:
Good point. Fixed!

inexorabletash reviewed (commented):
Thanks for submitting the test!

inexorabletash commented:
> Please ask @ayuishii for a general review here. If I'm still in some OWNERS file, please replace me with @ayuishii :)

Looks like you're still in `META.yml`

RupinMittal committed:
Cookie Store: Add testing for checks on path and domain length when setting cookies

The spec (https://wicg.github.io/cookie-store/#set-cookie-algorithm) for the
Cookie-Store API dictates that in the set function, if the byte sequence length
of the path (in UTF8 format) is greater than the maximum attribute value size
(currently 1024 bytes according to
https://wicg.github.io/cookie-store/#cookie-maximum-attribute-value-size),
then the promise should be rejected with a TypeError. The same is true for the
domain. This patch adds testing for this check in the set function in the same
file that tests other argument-related checks that the set function should be
doing.

RupinMittal commented:
> Please ask @ayuishii for a general review here. If I'm still in some OWNERS file, please replace me with @ayuishii :)

Do you know how I can request @ayuishii for a review. I replaced you with @ayuishii in `META.yml` but the system didn't request a review and I don't see a button to do that. Thank you!

inexorabletash commented:
LGTM! @pwnall since you requested changes can you mark them resolved so this can merge?
